web/src/search_pill.ts

```diff
@@ -34,14 +35,19 @@ type SearchPill =
           type: "search";
           display_value: string;
           description_html: string;
+          operator: string;
+          operand: string;
+          negated: boolean | undefined;
       }
     | SearchUserPill;
 
 export type SearchPillWidget = InputPillContainer<SearchPill>;
 
 export function create_item_from_search_string(search_string: string): SearchPill | undefined {
     const search_terms = Filter.parse(search_string);
-    if (!search_terms.every((term) => Filter.is_valid_search_term(term))) {
+    assert(search_terms.length === 1);
+    const search_term = search_terms[0]!;
+    if (!Filter.is_valid_search_term(search_term)) {
```

Comment:
This may change what happens if you paste some search terms in?

I guess testing this with pasting `sender:me has:image turtle` in and hitting enter, it seems to have the same (bugged!) behavior as pre-this-PR. So maybe worth investigation, but not a regression.

Reply:
Ok following up:

* You _can_ paste multiple pills at the same time, like `channel:denmark has:attachment`
* search terms like just `turtle` are never pillified, they stay as free text as we decided earlier
* `sender:me` specifically is causing an issue because it's an invalid recipient. "sender:me" shows the valid option in the dropdown, but actually pressing enter with "sender:me" parses "me" as an email and makes it invalid. I think this isn't great, but also doesn't feel super urgent to fix.

Reply:
Yeah let's just file an issue for "sender:me" not working with paste but not work on it immediately.

Reply:
#31315